x-pack/plugins/apm/public/components/shared/Links/url_helpers.ts

```diff
@@ -127,6 +127,10 @@ interface APMQueryParams {
   sortDirection?: string;
   sortField?: string;
   kuery?: string;
+  rangeFrom?: string;
+  rangeTo?: string;
+  refreshPaused?: string | boolean;
```

Comment:
A query param is always a string afaik. Any booleans would be stringified like `"true"`

Reply:
Right, but if I leave this as string only, then I have to manually convert `true` to `"true"` before I use `fromQuery({ refreshPaused: true })`. Since this interface is used for the converted query object, sometimes those values are strings when they come back from `qs.parse` and sometimes they're boolean when they are going in. 

Do you have thoughts on a better way to handle this?

Reply:
Yes, I suggest we fix `fromQuery` and `toQuery` in url_helpers. They are incorrect, and I never noticed until now. Something like this:


```ts
// toQuery always return an object of strings
export function toQuery(search?: string): APMQueryParamsRaw & RisonEncoded {
  return search ? qs.parse(search.slice(1)) : {};
}

// fromQuery is more liberal in what it accepts since it both accepts: `refreshPaused: true` and `refreshPaused: 'true'`
export function fromQuery(query: APMQueryParams & RisonEncoded) {
  const encodedQuery = encodeQuery(query, ['_g', '_a']);
  return stringifyWithoutEncoding(encodedQuery);
}

interface APMQueryParamsRaw {
  transactionId?: string;
  traceId?: string;
  detailTab?: string;
  flyoutDetailTab?: string;
  waterfallItemId?: string;
  spanId?: string;
  page?: string;
  sortDirection?: string;
  sortField?: string;
  kuery?: string;
  rangeFrom?: string;
  rangeTo?: string;
  refreshPaused?: string;
  refreshInterval?: string;
}

export interface APMQueryParams {
  transactionId?: string;
  traceId?: string;
  detailTab?: string;
  flyoutDetailTab?: string;
  waterfallItemId?: string;
  spanId?: string;
  page?: string | number;
  sortDirection?: string;
  sortField?: string;
  kuery?: string;
  rangeFrom?: string;
  rangeTo?: string;
  refreshPaused?: string | boolean;
  refreshInterval?: string | number;
}
```

Reply:
It's a bit of a mess right now with all the Rison. I hope we can move that out, and only handle that in the Discover Links.